app/node_modules/dimensions/terrariaserverpackethandler.ts

```diff
@@ -571,6 +612,23 @@ class TerrariaServerPacketHandler {
   private restoreVisuals(client: Client): void {
     client.player.setVisuals();
   }
+
+  private syncProjectiles(): void {
+    const projectileSync = this.currentServer.client.options.synchronization.projectiles;
+    if (projectileSync.enabled && (projectileSync.when === "always" || this.currentServer.isSSC)) {
+      const projectileIds = Array.from(this.currentServer.client.projectileTracking.keys())
+      projectileIds.forEach(projectileId => {
+        if (projectileSync.type === "clear") {
```

Comment:
It might be more efficient to pull this `if` higher, maybe even merge with the previous `if`.
Unless you want to, eventually, do something with the projectiles outside the clear case.

Reply:
there's another case "keep" that will be used for the opposite of clearing, which will update the server with the current client projectiles. Not sure whether this should be the default yet if not clearing.